extension/content/components/recipes/form/arguments/ShowHeartBeat.js

```diff
@@ -61,21 +61,35 @@ function RepeatOptionsField() {
     },
   ];
 
-  const repeatsEveryChangeSideEffect = ({ data, value }) => {
+  const repeatOptionChangeSideEffect = ({ data, value }) => {
     if (value !== "xdays") {
       /* eslint-disable no-unused-vars */
       const {
-        repeatEvery: _omitRepeatOption,
+        repeatEvery: _omitRepeatEvery,
         action,
         ...cleanedArguments
       } = data.arguments;
       /* eslint-enable no-unused-vars */
       return { ...data, arguments: cleanedArguments };
     }
 
-    return data;
+    return { ...data, arguments: { ...data.arguments, repeatEvery: 1 } };
```

Comment:
nit: order matters here so it should technically be `{ repeatEvery: 1, ...data.arguments }`